src/app/ModalRenderer/ModalRenderer.tsx

```diff
@@ -24,6 +26,12 @@ export const ModalRenderer: React.FC = (): ReactElement | null => {
     const notificationContent = (): string => {
         return selectedNode.hasEnabledNotifications ? "Disable" : "Allow";
     };
+    const handleRemoveNode = (): void => {
+        dispatch(submitDeleteNode(selectedNode.id));
+        dispatch(removeConfirmationDialog());
+        dispatch(resetAppState());
```

Comment:
I think you can have 1 action that could dispatch `submitDeleteNode`, `resetAppState` and `resetNodeList`.

Reply:
Or even better, dispatch them from `submitDeleteNode` after success. Removing confirmation dialog can stay here in the component as it is not data management but rendering thing.